python/rrc_simulation/sim_finger.py

```diff
@@ -294,12 +294,27 @@ def _get_latest_observation(self):
             # self.__applied_torque does not exist), set it to zero
             observation.torque = np.zeros(len(observation.velocity))
 
-        finger_tip_states = pybullet.getJointStates(
-            self.finger_id, self.pybullet_tip_link_indices
-        )
-        observation.tip_force = np.array(
-            [np.linalg.norm(tip[2][:3]) for tip in finger_tip_states]
-        )
+        finger_contact_states = [
+            pybullet.getContactPoints(
+                bodyA=self.finger_id,
+                linkIndexA=tip,
+                )
+            for tip in self.pybullet_tip_link_indices]
+        tip_forces = []
+        for i in range(len(finger_contact_states)):
+            directed_contact_force = 0.0
+            try:
+                for contact_point in range(len(finger_contact_states[i])):
```

Comment:
would this work as well?
```suggestion
                for contact_point in finger_contact_states[i]:
```
and replacing `finger_contact_states[i][contact_point]` with `contact_point` below?